00_Base/src/services/credentials.service.ts

```diff
@@ -542,6 +532,111 @@ export class CredentialsService {
     );
   }
 
+  async regenerateCredentialsToken(
```

Comment:
@lydiazcheng this method looks good, and the comments are very helpful. For readability, can you please create some smaller helper methods to break this larger method down, which should also help with testing 🙏

Reply:
Thanks Elliot! I moved the detailed code into helper functions and only kept the code that call these new functions in the `regenerateCredentialsToken` method.  Please let me know if I should cut more from this method.

Reply:
exactly, this is great - thank you!